docs/docs/platform/step-filter.md

```diff
@@ -129,3 +135,42 @@ if (hmacHash === hmac) {
 The webhook should return a one-level object containing any information that you want to use
 to filter the notification flow. For example, you might include a field called 'isOnline' that
 indicates whether the subscriber is currently online.
+
+## Subscriber online filters
+
+This feature allows you to apply a filter on step notifications based on the subscriber's online status. The filter can be applied to determine if the subscriber is currently online or if the subscriber was online in the "X" time-period.
+
+Example:
+To send an email notification to all subscribers who were active within the last 30 minutes, follow these steps:
+
+- On the Workflow editor page, Add the "Email" step notification.
+- Click on the "Add Filter" button from the sidebar.
+- Select "Online in the last X time period" and choose "minutes" as the time period.
+- Enter "30" in the input field.
+
+By applying this filter, the notification will only be delivered to subscribers who have been active within the last 30 minutes, ensuring that the message is reaching engaged and active recipients.
+
+### Online filter Mechanism
+
+Novu uses a websocket connection within the `notification-center` package to track the online status of subscribers. The `isOnline` and `isOnlineInLast` fields of the subscriber's entity are updated accordingly.
+
+When a subscriber comes online, an active websocket connection is established with the server. Novu then updates the isOnline and `isOnlineInLast` fields of the subscriber's entity to `isOnline: true` and `isOnlineInLast: current_timestamp`. When the subscriber disconnects, Novu updates`isOnline: false` and `isOnlineInLast: current_timestamp`.
+
+The online filter feature can be used to determine if a subscriber is online right now or if the subscriber was online within a specific time period.
+
+- To determine if a subscriber is online right now, Novu checks the value of the `isOnline` field. If `isOnline` is `true`, the subscriber is online, otherwise the subscriber is offline.
```

Comment:
", otherwise when the `isOnline` is `false` the subscriber is considered offline."

Not sure if we should say that the subscriber might not have these fields set at the beginning, and if not we won't deliver a message, because we don't know whether he is online or not.

Reply:
I think once a subscriber is added to the system and it might not have those fields, technically the subscriber should be considered offline, until the subscriber logs in.

Reply:
@BiswaViraj I don't think that we can say that, because actually, it could be online at that time, so we don't know.